logstash-core/lib/logstash/runner.rb

```diff
@@ -122,10 +121,8 @@ def execute
   rescue LoadError => e
     fail("Configuration problem.")
   rescue LogStash::ConfigurationError => e
+    @logger.warn I18n.t("logstash.runner.configtest-flag-information")
     @logger.fatal I18n.t("logstash.error", :error => e)
-    if !config_test?
-      @logger.warn I18n.t("logstash.runner.configtest-flag-information")
-    end
     show_short_help
     return 1
```

Comment:
Clamp will return the proper exit code. This is really useful! The reason being that if I called exit directly this module wouldn't really be testable.